parlai/scripts/convo_render.py

```diff
@@ -0,0 +1,241 @@
+                conversation += [(speaker, item["text"])]
+            conversation += [(END_OF_CONVO, END_OF_CONVO)]
+
+    return conversation
+
+
+def prBlueBG(text):
+    """
+    Print given in text with a blue background
+    :param text: The text to be printed
+    """
+    print("\033[44m{}\033[0m".format(text), sep="")
+
+
+def display_cli(conversations, alt_speaker, human_speaker):
+    """
+    Display the conversations on the Command Line
+    :param conversations: The dialogs to be displayed
+    :param alt_speaker: Name of other speaker to be used
+    :param human_speaker: Name of human speaker to be used
+    """
+    for speaker, speech in conversations:
+        if speaker == END_OF_CONVO:
+            print("-" * 20 + "END OF CONVERSATION" + "-" * 20)
+        elif speaker == alt_speaker:
+            print("%-15s: %s" % (speaker[:15], speech))
+        else:
+            prBlueBG("%-15s: %s" % (speaker[:15], speech))
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description="Process Conversation Rendering arguments"
+    )
+    parser.add_argument("i", help="Input file to read conversations from")
+    parser.add_argument(
+        "-o",
+        help="Output file to write conversations to. One of [.pdf, .png, .html] only",
+    )
+    parser.add_argument("-wd", help="Width of output file", type=int, default=8)
+    parser.add_argument("-ht", help="Height of output file", type=int, default=9.5)
+    parser.add_argument(
+        "-ne", help="Number of conversations to render", type=int, default=10
+    )
+
+    args = parser.parse_args()
+    input_file, output_file = args.i, args.o
+    height, width = args.ht, args.wd
+    alt_speaker = input_file.split('/')[-1][:-6]
+
+    dialogs = pre_process(input_file, args.ne, alt_speaker)
+
+    if output_file is None:
+        # CLI
+        display_cli(dialogs, alt_speaker, "human")
+    else:
+        extension = output_file.split(".")[-1]
+        if extension not in ["html", "pdf", "png"]:
+            raise Exception(
+                "Extension not specified/supported. Specify one of '.html', '.pdf' or '.png' output files"
+            )
+        html_str = gen_html(
+            dialogs, height, width, "Rendered HTML", alt_speaker, "human"
+        )
+        if extension == "html":
+            # save to output
+            file_handle = open(output_file, "w")
+            file_handle.write(html_str)
+            file_handle.close()
+        else:
+            # create temp dir
+            with tempfile.TemporaryDirectory() as tmpdir:
+                fname = tmpdir + "/interim.html"  # save html to interim.html in tmpdir
+                file_handle = open(fname, "w")
+                file_handle.write(html_str)
+                if extension == "pdf":
+                    cmd = (
+                        '/Applications/Google\\ Chrome.app/Contents/MacOS//Google\\ Chrome --headless --crash-dumps-dir=/tmp --print-to-pdf="'
```

Comment:
f-string

Reply:
Also pull out the path to Chrome as a constant. We'll need this to run `chrome-browser` on linux